scenarios.md

```diff
@@ -0,0 +1,225 @@
+  `$ docker tag net-monitor:abc123 wabbitnetworks.example.com/networking/net-monitor:1.0`  
+  `$ docker push wabbitnetworks.example.com/networking/net-monitor:1.0`
+1. Deploy the artifact to a cluster that requires signatures:  
+  `$ orchestrator apply -f deploy.yaml`
+1. The orchestrator in the dev environment accepts any signed content, enabling it to trace where deployed artifacts originated from.
+
+**Implications of this requirement:**
+
+- Signatures can be verified based on the referenced `:tag`. The signature is linked to a unique manifest, and not tied to a specific `repo:tag` name. 
+- The artifact can be renamed from the unique build id `net-monitor:abc123` to a product versioned tag `wabbitnetworks.example.com/networking/net-monitor:1.0` without invalidating the signature.
+- Users may reference the `sha256` digest directly, or the `:tag`. While tag locking is not part of the [OCI Distribution Spec][oci-distribution], various registries support this capability, allowing users to reference human readable tags, as opposed to long digests. Either reference is supported with Notary v2, however it's the digest that is signed.
+- Notary v2 supports a pattern for signing any type of artifact, from OCI Images, Helm Charts, Singularity to yet unknown types.
+- Orchestrators may require signatures, but not enforce specific specific signatures. This enables a host to understand what content is deployed, without having to manage specific keys.
+
+### Scenario #3: Automate Build, Sign, Push, Deploy to Prod, Verify
+
+A CI system is triggered by a git commit. The system builds the artifacts, signs them, and pushes to a registry. The production system pulls the artifacts, verifies the signatures and runs them.
+
+1. A CI solution responds to a git commit notification
+1. The CI system clones the git repo and builds the artifacts, with fully qualified names:  
+  **image**: `wabbitnetworks.example.com/networking/net-monitor:1.0-alpine`   
+  **deployment chart**: `wabbitnetworks.example.com/networking/net-monitor:1.0-deploy`
+1. The CI system signs the artifact with locally available keys.
+1. The CI system creates a signed OCI Index, referencing the image and deployment charts:  
+  `wabbitnetworks.example.com/networking/net-monitor:1.0`
+1. The index, and its contents are pushed to a registry:  
+  `$ docker push wabbitnetworks.example.com/networking/net-monitor:1.0-alpine`  
+  `$ deploy-tool push wabbitnetworks.example.com/networking/net-monitor:1.0-deploy`  
+  `$ oci-tool push wabbitnetworks.example.com/networking/net-monitor:1.0`
+1. The artifacts are deployed to a production orchestrator.
+1. The orchestrator verifies the artifacts are signed by a set of specifically trusted keys. Unsigned artifacts, or artifacts signed by non-trusted keys are rejected.
+
+**Implications of this requirement:**
+
+- Keys for signing are securely retrieved by build systems that create & destroy the environment each time.
```

Comment:
Hmm... is the ability of access to keys here mean actual access to key bytes? Or ability to sign with a particular key?

I think the ability for key bytes to be explicitly be accessible is restrictive. My thought process is around key based protocols like KMIP are moving towards ability to sign something remotely. And the same would be for PKCS11 for HSMs (and services built around it). 

The implication for this would be signatures stored require support for a variety of standard formats, like PKCS7. Or have some library that has the ability to move signatures in/out of the format of notaryv2 and pkcs7.

Reply:
The main point I was trying to make here was private signing keys are accessible by external entities, and not directly part of the notary workflow. Nor, should we assume a developers workstation, or build system must be heavily configured. Details on the types of keys and such, I've left out of the scenario.
This also hints at the vendor agnostic components as well. We had a cloud/vendor requirement that allows the workflows to take advantage of the vendor/cloud providers keyvault solution. 
Is there a suggestion for how I can reword this to not imply a constraint on how things are signed?

Reply:
Hmm... how about  "The CI system performs signing of the artifact via locally available keys or through a remote signing services"?